internal/pkg/task/task.go

```diff
@@ -54,18 +55,43 @@ func taskFamilyName(groupName string) string {
 	return fmt.Sprintf(fmtTaskFamilyName, groupName)
 }
 
-func newTaskFromECS(ecsTask *ecs.Task) *Task {
+func newTaskFromECS(ecsTask *ecs.Task) (*Task, error) {
+	taskARN := aws.StringValue(ecsTask.TaskArn)
+	eni, err := ecsTask.ENI()
```

Comment:
let's simplify I think all this error handling logic and ignore the error if we fail to get the ENI since it's strictly additional information and not required to create the tasks.

```suggestion
	eni, _ := ecsTask.ENI() //  Best-effort parse the ENI. If we can't find an IP address, we won't show it to the customers instead of erroring.
```